internal/dinosaur/pkg/services/quota/ams_quota_service.go

```
@@ -235,3 +217,46 @@ func (q amsQuotaService) DeleteQuota(subscriptionID string) *errors.ServiceError
 	}
 	return nil
 }
+
+func mapAllowedQuotaCosts(quotaCosts []*amsv1.QuotaCost) (map[amsv1.BillingModel][]*amsv1.QuotaCost, error) {
+	costsMap := make(map[amsv1.BillingModel][]*amsv1.QuotaCost)
+	var foundUnsupportedBillingModels []string
+	for _, qc := range quotaCosts {
+		// When an SKU entitlement expires in AMS, the allowed value for that quota cost is set back to 0.
+		// If the allowed value is 0 and consumed is greater than this, that denotes that the SKU entitlement
+		// has expired and is no longer active.
+		if qc.Allowed() == 0 {
+			continue
+		}
+		for _, rr := range qc.RelatedResources() {
+			bm := amsv1.BillingModel(rr.BillingModel())
+			if _, isCompatibleBillingModel := supportedAMSBillingModels[rr.BillingModel()]; isCompatibleBillingModel {
+				costsMap[bm] = append(costsMap[bm], qc)
+			} else {
+				foundUnsupportedBillingModels = append(foundUnsupportedBillingModels, rr.BillingModel())
+			}
+
+		}
+	}
+	if len(costsMap) == 0 && len(foundUnsupportedBillingModels) > 0 {
```

Why do we fail if we find an unsupported billing model? Can we just ignore them? I think that would be more error-proof approach considering quota configuration is external to `fleet-manager`

---

We fail because there's no supported model. We decide on entitlement based on that, so I'd rather not swallow errors.

---

So the error then is not that there is unsupported billing model, but rather that there are no supported billing models with active SKU entitlement. Let's make error message reflect that

---

Not addressed